src/lib/encoder/Encoder.h

```diff
@@ -99,14 +100,34 @@ inline unsigned Encoder::encode(const std::string& filename, std::string output_
 	return i;
 }
 
-inline unsigned Encoder::encode_fountain(const std::string& filename, const std::function<bool(const cv::Mat&, unsigned)>& on_frame)
+inline unsigned Encoder::encode_fountain(const std::string& filename, const std::function<bool(const cv::Mat&, unsigned)>& on_frame, int compression_level)
 {
-	std::ifstream f(filename);
-	fountain_encoder_stream fes = fountain_encoder_stream::create(f, cimbar::Config::fountain_chunk_size(_eccBytes), 0); // will eventually do something clever with encode_id?
+	unsigned chunk_size = cimbar::Config::fountain_chunk_size(_eccBytes);
+
+	std::stringstream ss;
+	std::ifstream infile(filename);
+	if (compression_level <= 0)
+		ss << infile.rdbuf();
+	else
+	{
+		cimbar::zstd_compressor<std::stringstream> f;
+		if (!f.compress(infile))
+			return 0;
+
+		// find size of compressed zstd stream, and pad it if necessary.
+		size_t compressedSize = f.size();
+		if (compressedSize < chunk_size)
+			f.pad(chunk_size - compressedSize + 1);
+		ss = std::move(f);
+	}
+
+	fountain_encoder_stream fes = fountain_encoder_stream::create(ss, chunk_size, 0); // will eventually do something clever with encode_id?
 	// With ecc = 40, we have 60 rs blocks * 115 bytes per block == 6900 bytes to work with.
 	// the fountain_chunk_size will be 690.
 	// fountain_chunks_per_frame() is currently a constant (10).
 	unsigned requiredFrames = fes.blocks_required() * 2 / cimbar::Config::fountain_chunks_per_frame();
+	if (requiredFrames == 0)
+		requiredFrames = 1; // could also do +1 on the division above?
```

Comment:
The `requiredFrames` calculation in general could use some tweaking -- for `cimbar_video`, more frames is better. (and 2x or even 3 or 4 or 5x are all fine). But for outputting images to file -- right now just for testing purposes, but also presumably for the "print to paper" case -- we really shouldn't need more than a handful of extra frames.